### ISSUE TITLE
Error 400 while trying to access a shared drive item content

### ISSUE THREAD

**Issue by jcze91** [microsoftgraph/msgraph-sdk-dotnet#947]
**Describe the bug**
Hello,
I'm trying to access a shared drive item via Microsoft Graph API by using the C# SDK.
It's ok sometimes, but randomly it's failing on 400 error.

**To Reproduce**
Here is my code : 
` var graphClient = new GraphServiceClient(CreateAuthProvider());
                    var token = $"u!{Convert.ToBase64String(Encoding.UTF8.GetBytes(url)).TrimEnd('=').Replace('/', '_').Replace('+', '-')}";
                    var content = await graphClient.Shares[token].DriveItem.Content.Request(new List<Option> { new HeaderOption("Prefer", "redeemSharingLink") }).GetAsync();
`

**Expected behavior**
Obtaining the drive item content

**Desktop (please complete the following information):**
 - OS: Windows
 
**Additional context**
I'm using the SDK C# version 3.28.0.

**Comment by MIchaelMainer** [comment]
Do you retry on that 400 status code and if so, does it successfully return? Does this error occur when trying to access the same item?

**Comment by jcze91** [comment]
No, there isn't retry on 400 error and we're not trying to access the same item.
For additional context, we try to fetch multiple drive items (10 to 20)  consecutively, and each time we time to reproduce it, the 400 error is on a different item.
I'll try to fetch them via one batch Graph API call

**Comment by jcze91** [comment]
When trying with batch API call, I still have some request ending in 400 error Bad Request, so I suppose it's not a concurrency problem.

I could found a workaround by using a retry policy on the call, but it's not a very suitable patch.
`                var graphClient = new GraphServiceClient(CreateAuthProvider());
                var token = $"u!{Convert.ToBase64String(Encoding.UTF8.GetBytes(url)).TrimEnd('=').Replace('/', '_').Replace('+', '-')}";
                var content = await Policy.Handle<ServiceException>(ex =>
                    ex.StatusCode == HttpStatusCode.BadRequest)
                    .Or<HttpRequestException>()
                   .WaitAndRetryAsync(7, retryCount => TimeSpan.FromSeconds(Math.Pow(2, retryCount) / 2))
                   .ExecuteAsync<Stream>(() => graphClient.Shares[token].DriveItem.Content.Request().GetAsync());`

For information, this bug looks to have occurred only in the last few days, since the 30th March on our side.
